GTiff: allow customising overview compression type (fixes #3453) (#3487)
OSGeo/gdal: gdal/doc/source/drivers/raster/cog.rst

        available if general options (i.e. options which are not creation options,
        like subsetting, etc.) are used.

> - **OVERVIEW_COMPRESS**=[AUTO/NONE/LZW/JPEG/DEFLATE/ZSTD/WEBP/LERC/LERC_DEFLATE/LERC_ZSTD]**:

Reviewer: ```suggestion
- **OVERVIEW_COMPRESS=[AUTO/NONE/LZW/JPEG/DEFLATE/ZSTD/WEBP/LERC/LERC_DEFLATE/LERC_ZSTD]**:
```

- **ADD_ALPHA=YES/NO**: Whether an alpha band is added in case of reprojection.
  Defaults to YES.

Configuration options
---------------------

> COG Overview creation can be configured with the :ref:`raster.gtiff` driver creation options

Reviewer: ```suggestion
COG Overview creation can be configured with the :ref:`raster.gtiff` driver configuration options (typically specified with ``--config NAME VALUE`` syntax)
```

> COG Overview creation can be configured with the :ref:`raster.gtiff` driver configuration options (typically specified with ``--config NAME VALUE`` syntax)

Reviewer 2: ```suggestion
By default the COG driver creates overviews with the same compression method than the full resolution image. The default behavior can be changed with the :ref:raster.gtiff driver configuration options (typically specified with --config NAME VALUE syntax).
```
